src/main/java/map/Point.java

```diff
@@ -13,12 +13,29 @@ public Integer to1D(Integer size) {
     if ((x >= size) || (y >= size)) {
       throw new IndexOutOfBoundsException("Battlefield map size should NOT be >= X OR  >= Y");
     }
-    
+
     return (size * x) + y;
   }
-  
+
   @Override
   public String toString() {
     return String.format("(X:%d, y:%d)", x, y);
   }
+
+  @Override
+  public boolean equals(Object point) {
```

Comment:
The implementation is too complicated. This makes it hard to read/understand. Rewrite it.